### DISCUSSION TITLE
Question for supporting RPC mechanism(request/reply communication paradigm)

### DISCUSSION THREAD

**Discussion by keepgoingxxx** [OpenDDS/OpenDDS#4182] [Q&A]
Hello,
I'm just starting out with DDS.

Does opendds support RPC mechanism(request/reply communication paradigm)?
If yes, can you please share an example?

Thank you in advance.

**Comment by mitza-oci** [comment]
OpenDDS doesn't have an implementation of the RPC over DDS specification https://www.omg.org/spec/DDS-RPC

However, that doesn't prevent an application from communicating requests and replies over normal DDS Topics.

Also note that once you've built OpenDDS you have also built TAO which has a mature, standards conforming, performant, and complete RPC solution.
